### PR TITLE
Lit EVM and SOL Wallet Clients

### PR DESCRIPTION
**Pull request by spacesailor24**
- Lit package [README](https://github.com/goat-sdk/goat/blob/61228d6c790b267336db716c4e3a6791c7421920/typescript/packages/wallets/lit-protocol/README.md)
- Example [README](https://github.com/goat-sdk/goat/blob/37ad656fa2c10d00037399c7ead26b48538fb7a2/typescript/examples/langchain/lit/README.md)
- LitEVMWalletClient [example](https://github.com/goat-sdk/goat/blob/37ad656fa2c10d00037399c7ead26b48538fb7a2/typescript/examples/langchain/lit/src/evm.ts)
- LitSolanaWalletClient [example](https://github.com/goat-sdk/goat/blob/37ad656fa2c10d00037399c7ead26b48538fb7a2/typescript/examples/langchain/lit/src/sol.ts)

### CODE REVIEW

File: typescript/packages/wallets/viem/src/index.ts [hunk 1]
```diff
@@ -62,7 +62,10 @@ export function viem(client: ViemWalletClient, options?: ViemOptions): EVMWallet
             if (!client.account) throw new Error("No account connected");
 
             const signature = await client.signTypedData({
-                domain: data.domain,
+                domain: {
+                    ...data.domain,
+                    chainId: typeof data.domain.chainId === 'bigint' ? Number(data.domain.chainId) : data.domain.chainId,
+                },
```
Comment (spacesailor24) [inline typescript/packages/wallets/viem/src/index.ts:68]: Without this a type error was being thrown:

```
src/index.ts(65,17): error TS2322: Type 'TypedDataDomain' is not assignable to type '{ chainId?: number | undefined; name?: string | undefined; salt?: `0x${string}` | undefined; verifyingContract?: `0x${string}` | undefined; version?: string | undefined; }'.
@goat-sdk/wallet-viem:build:   Types of property 'chainId' are incompatible.
@goat-sdk/wallet-viem:build:     Type 'number | bigint | undefined' is not assignable to type 'number | undefined'.
@goat-sdk/wallet-viem:build:       Type 'bigint' is not assignable to type 'number'.
```

**Review by 0xaguspunk — COMMENTED** [review]
I know it's still WIP! made some small comments and asked a few questions :)

Looking good tho!

File: typescript/packages/core/src/wallets/lit.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    litEVMChainIdentifier: string;
+    viemWalletClient: WalletClient;
+};
+ 
+export type LitSolanaWalletOptions = LitWalletOptions & {
+    network: "solana";
+};
+
+export interface LitEVMWalletClient extends EVMWalletClient {
+    getWrappedKeyMetadata(): Promise<StoredKeyData>;
```
[the remark this answers is not in the document]
Comment (spacesailor24) [reply typescript/packages/core/src/wallets/lit.ts]: I've removed this in [this](https://github.com/goat-sdk/goat/pull/43/commits/2c3153ecfb56081d4f8a6ea0545634342e625f97) commit

File: typescript/packages/wallets/lit-protocol/README.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# Goat Wallet Lit Protocol 🐐 - TypeScript
+
+## Installation
+```
+npm install @goat-sdk/wallet-lit
+```
+
+## Usage
+```typescript
+
```
[the remark this answers is not in the document]
Comment (spacesailor24) [reply typescript/packages/wallets/lit-protocol/README.md:125]: Added in [this commit](https://github.com/goat-sdk/goat/pull/43/commits/83ae31bb57ed537265c8053f702e0b00447bf3c5)

File: typescript/packages/plugins/uniswap/package.json [hunk 4]
```diff
@@ -22,7 +22,7 @@
     "dependencies": {
         "@goat-sdk/core": "workspace:*",
         "viem": "^2.21.49",
-        "zod": "^3.23.8"
+        "zod": "catalog:"
```
Comment (spacesailor24) [inline typescript/packages/plugins/uniswap/package.json:25]: This was causing a mismatch error between used zod versions

**Pull request merged**